integration-tests/smoke/keeper_test.go

```diff
@@ -171,33 +165,41 @@ func getKeeperSuite(
 
 		Describe("with Keeper job", func() {
 			if testToRun == BasicSmokeTest {
-				It("performs upkeep of a target contract, stops upon cancel", func() {
-					// Let upkeep be performed atleast once
-					Eventually(func(g Gomega) {
-						cnt, err := consumer.Counter(context.Background())
-						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
-						g.Expect(cnt.Int64()).Should(BeNumerically(">", int64(0)), "Expected consumer counter to be greater than 0, but got %d", cnt.Int64())
-						log.Info().Int64("Upkeep counter", cnt.Int64()).Msg("Upkeeps performed")
-					}, "1m", "1s").Should(Succeed())
-
-					// Now cancel the upkeep as registry owner, it should get immediately cancelled
+				It("monitors all the registered upkeeps perform and checks the cancel functionality for the last of them", func() {
+					for i := 0; i < len(upkeepIDs); i++ {
```

Comment:
you can put the for loop inside eventually also

Reply:
imo it's more clearer semantics if we have loop inside. Eventually within 1 minute all upkeeps should have been performed once. 

With loop outside you'll be waiting for different times for different upkeeps since you'll be doing it sequentially. 

Having the loop inside also makes it clear that the test will wait for 1 minute. otherwise the test can theoretically end up waiting for upto 10 minutes

Reply:
That makes a lot of sense...especially the last advantage. Thanks a lot for pointing it out!